dashboard/app/dsl/script_dsl.rb

```diff
@@ -76,21 +78,43 @@ def level_concept_difficulty(json)
 
   string :video_key_for_next_level
 
+  boolean :active
+
   def assessment(name)
     level(name, {assessment: true})
   end
 
   def level(name, properties = {})
-    @levels << {
+    level = {
       :name => name,
-      :stage => @stage,
       :stage_flex_category => @stage_flex_category,
       :skin => @skin,
       :concepts => @concepts.join(','),
       :level_concept_difficulty => @level_concept_difficulty || {},
       :video_key => @video_key_for_next_level
     }.merge(properties).select{|_, v| v.present? }
     @video_key_for_next_level = nil
+    if @current_scriptlevel
+      @current_scriptlevel[:levels] << level
+      if !@active
+        @current_scriptlevel[:properties][name] = { :active => @active }
+        @active = true
+      end
+    else
+      @scriptlevels << {
+        :stage => @stage,
+        :levels => [level]
+      }
+    end
+  end
+
+  def variants
+    @current_scriptlevel = { :levels => [], :properties => {}, :stage => @stage}
+  end
+
+  def endvariants
+    @scriptlevels << @current_scriptlevel
+    @current_scriptlevel = nil
```

Comment:
I'm wondering if it's worth making using a more conventional DSL format here by replacing `variants`/`endvariants` methods with a single block-based method `variants(&block)`. For example:
```ruby
stage 'Stage1'
level 'Level 1'
variants {
  level 'Level 2a'
  active false
  level 'Level 2b'
}
level 'Level 3'
```
though if we really wanted to do this 'right', we should replace all other stateful methods like `stage` with block parameters as well, e.g.:
```ruby
stage 'Stage1' {
  level 'Level1'
  variants {
    level 'Level 2a'
    level 'Level 2b', active: false
  }
}
```
For this reason maybe it's worth making such a change later as a bigger refactor, since we have a bunch of existing stateful methods like this that we would want to change (and mass-migrate all of our existing script DSL files to) for consistency sake.

@joshlory thoughts? I know in the past we had discussed possibly making the Script DSL obsolete at some point anyway, so maybe not even worth investing even more effort into a clean design here?